### PR TITLE
filemanager: setup testing

### LINKED ISSUE #71: filemanager: setup testing mocks
Since we are removing localstack, we should include mock objects to do local tests before deploying to dev. Ideally, it should be possible to run `cargo test` to quickly check unit tests. The docker postgres database can be used as a local database connection, or sqlx already has [some testing](https://docs.rs/sqlx/latest/sqlx/attr.test.html) functionality. However, the AWS clients will need to be mocked to create unit tests, similar to how it's described [here](https://docs.aws.amazon.com/sdk-for-rust/latest/dg/testing.html). Currently, this includes `aws_sdk_sqs::Client` and `aws_sdk_s3::Client`.

### PR DESCRIPTION
**Pull request by mmalenic**
Closes #71

### Changes
* Adds database and unit tests.
* Adds AWS test mocks using [mockall](https://github.com/asomers/mockall).
* General refactor to increase clarify and update docs.
* Fix ordering and equality bug when de-duplicating and sorting events within the same Lambda function.

It should be a lot easier to test locally now just by running `cargo test`. This will mean that dev deployments are more likely to succeed if the tests pass locally. Now, if only there is way to remove `docker compose up` as well. Just being able to run `cargo test` directly would be the most convenient.

### CODE REVIEW

File: lib/workload/stateful/filemanager/filemanager/src/clients/aws/config.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+use aws_config::{defaults, BehaviorVersion, ConfigLoader, SdkConfig};
+use mockall::automock;
+
+/// A wrapper around a config loader.
+#[derive(Debug)]
+pub struct Config {
+    inner: ConfigLoader,
+}
+
+#[automock]
```
Comment (mmalenic) [inline lib/workload/stateful/filemanager/filemanager/src/clients/aws/config.rs:10]: This creates an entirely new `MockConfig` struct which has methods that allow setting expectations.

File: lib/workload/stateful/filemanager/filemanager/src/events/aws/collecter.rs [hunk 2]
```diff
@@ -0,0 +1,246 @@
+#[double]
```
Comment (mmalenic) [inline lib/workload/stateful/filemanager/filemanager/src/events/aws/collecter.rs:1]: This expands to:
```rust
#[cfg(not(test))]
use crate::clients::aws::s3::Client;
#[cfg(test)]
use crate::clients::aws::s3::MockClient as Foo;
```
Which allows using different versions of the client in testing code compared to the main code.

**Review by brainstorm — APPROVED** [review]
Very nice use of mockall and mock_double, I didn't know those crates, kudos Marko!

**Comment by mmalenic** [comment]
Do you want to go for #74 first? There might be some conflicts.

**Comment by brainstorm** [comment]
> Do you want to go for #74 first? There might be some conflicts.

Ok, I'll put it up for review (now it's in draft), but there's still things left to fix :_S

**Pull request merged**